### ISSUE TITLE
Sort bookmarks in more popular order

### ISSUE THREAD

**Issue by bhajneet** [ShabadOS/desktop#150]
By and far the most used bookmark is `6 pauria(n) anand sahib with salok`. This was at the top of the bookmark list in v1 golang. If the 6 lines shown in the bookmark screen were all popular banis, it would make it easier for users to know what good options are there quicker. Here are some more popularly used bookmarks:

6 pauri Anand sahib, Asa di vaar, rehraas, kirtan sohila, chaupai sahib, and anand karaj

**Comment by Harjot1Singh** [comment]
We could just re-order it in the DB and that'd be reflected?

**Comment by bhajneet** [comment]
Up to you, I have no concerns

**Comment by Sarabveer** [comment]
That should be removed from DB as its not really a `Baani` and added as a Desktop only bookmark.

**Comment by Harjot1Singh** [comment]
Not sure how to facilitate the idea of bookmarks/compilations without renaming the table tbh, adding to desktop-only is doable but a bit of a pain because it means we’d be storing data concerns in another place too

**Comment by Harjot1Singh** [comment]
Maybe we could rename the table and introduce a bani Boolean field

**Comment by Sarabveer** [comment]
> Maybe we could rename the table and introduce a bani Boolean field

That could possibly work

**Comment by Sarabveer** [comment]
> We could just re-order it in the DB and that'd be reflected?

Remember DB has to cater to more than just Shabad OS Desktop. Current order reflects a gutka.

**Comment by bhajneet** [comment]
At some point there will be customizable bookmarks. DB can have it sorted by gutka or more ideally in alphabetical order as a DB should. It is up to the client how to order them and which ones to hide/show imo.

Maybe you can break it up into non-customizable bookmarks (hardcoded, required bookmarks) and client-side customized bookmarks. Still you'll need to be able to sort them however the user likes regardless of how it was shown in the db.

**Comment by Harjot1Singh** [comment]
Ok, client-side ordering do-able, what do we consider to be immutable fields? The name of a bani?



---
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

**Comment by bhajneet** [comment]
This would be immutable (Eng name, Gurmukhi name, Lines): https://github.com/ShabadOS/database/blob/master/data/banis.json

Custom bookmarks the user adds (for example let's say "starring/favoriting" a shabad) can be deleted.

Not about the fields but the source of the bookmark. Embedded in our DB is immutable, added/created by user can be deleted.

**Comment by Harjot1Singh** [comment]
What I mean is, we’d have to store a reference on the client-side to these Banis (name, id, whatever we chose, preferably something that doesn’t change so that it can’t invalidate the client-side).

The bani ID can certainly change (unless we used immutable IDs like we do for Lines and Shabads), the next best thing I can think of is Bani name, which can change, but not often?


---
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus
